src/service/useLocalStorage.js

```diff
@@ -0,0 +1,26 @@
+import { useState } from "react";
+
+export function useLocalStorage(key, initialValue) {
+  const [storedValue, setStoredValue] = useState(() => {
+    try {
+      const item = localStorage.getItem(key);
+      return item ? JSON.parse(item) : initialValue;
```

Comment:
This is correct logic to get the initial value (one from localStorage, if not, then use `initialValue` (all good))

Ideally, this hook should always be in sync with the localStorage, right from the start

So what's missing here is - in case the initial value does not exist in localStorage yet (the `initialValue` is about to be set), this `initialValue` needs to be saved to localStorage.

So you could do something like this:

```js
// if the value exists in localStorage, return it
if (item) {
  return JSON.parse(item);
}

// if the value doesn't exist in localStorage, save initialValue to localStorage and return it
localStorage.setItem(key, JSON.stringify(initialValue));
return initialValue;
```

With this implementation, it is guaranteed that you will always have saved value or initial (defaultValue) value that you set

Now you don't have to use fallbacks like ` || {}` or ` || defaultSources`  anymore